bs/src/tablecloth.mli

```diff
@@ -1,68 +1,476 @@
+
+    ### Example
+
+    ```ocaml
+    let cube_root (x : int) =
+      ((float_of_int x) ** (1.0 /. 3.0) : float)
+      
+    map ~f:cube_root [8; 1000; 1728] (* [2; 9.999..; 11.999..] *)
+    ```
+
+    ```reason
+    let cube_root = (x:int): float => { float_of_int(x) ** (1.0 /. 3.0); }
+    map(~f=cube_root, [8, 1000, 1728]) /* [2, 9.999.., 11.999..] */
+    ```
+  *)
   val map : f:('a -> 'b) -> 'a list -> 'b list
 
+  (**
+    Same as indexed_map
+  *)
   val indexedMap : f:(int -> 'a -> 'b) -> 'a list -> 'b list
 
+  (**
+    `indexed_map ~f:fcn xs`{:.ocaml} `indexedMap(~f=fcn, xs)`{:.reason} returns a new list that it is the result of applying function `fcn` to each item in the list `xs`. The function has two parameters: the index number of the item in the list, and the item being processed. Item numbers start with zero.
+
+    ### Example
+
+    ```ocaml
+    let numbered (idx: int) (item: string) =
+      ((string_of_int idx) ^ ": " ^ item : string)
+
+    indexed_map ~f:numbered ["zero"; "one"; "two"] =
+      ["0: zero"; "1: one"; "2: two"]
+    ```
+
+    ```reason
+    let numbered = (idx: int, item: string): string =>
+      string_of_int(idx) ++ ": " ++ item;
+      
+    indexedMap(~f=numbered, ["zero", "one", "two"]) ==
+      ["0: zero", "1: one", "2: two"]
+    ```
+  *)
   val indexed_map : f:(int -> 'a -> 'b) -> 'a list -> 'b list
 
+  (**
+    Same as indexed_map
+  *)
   val mapi : f:(int -> 'a -> 'b) -> 'a list -> 'b list
 
+  (*
+    `map2 ~f: fcn xs ys`{:.ocaml} `map2(~f=fcn, xs, ys)`{:.reason} returns a new list whose items are `fcn x y`{:.ocaml} `fcn(x,y)`{:.reason} where `x` and `y` are the items from the given lists.
+
+    ### Example
+    ```ocaml
+    let discount (price: float) (percentage: float) =
+      (price *. (1.0 -. (percentage /. 100.0)) : float)
+      
+    map2 ~f:discount [100.0; 85.0; 30.0] [10.0; 20.0; 30.0] =
+      [90.0; 68.0; 21.0]
+    ```
+
+    ```reason
+    let discount = (price: float, percentage: float): float =>
+      price *. (1.0 -. (percentage /. 100.0));
+      
+    map2(~f=discount, [100.0, 85.0, 30.0], [10.0, 20.0, 30.0]) ==
+      [90.0, 68.0, 21.0]
+    ```
+  *)
   val map2 : f:('a -> 'b -> 'c) -> 'a list -> 'b list -> 'c list
 
+  (**
+    `getBy ~f: predicate xs`{:.ocaml} `getBy(~f=predicate, xs)`{:.reason} returns `Some value`{:.ocaml} `Some(value)`{:.reason} for the first value in `xs` that satisifies the `predicate` function; returns `None` if no element satisifies the function. 
+
+    ### Example
+
+    ```ocaml
+    let even (x: int) = (x mod 2 = 0 : bool)
+    getBy ~f:even [1;4;3;2]) = Some 4
+    getBy ~f:even [15;13;11]) = None
+    ```
+
+    ```reason
+    let even = (x: int): bool => {x mod 2 == 0};
+    getBy(~f=even, [1, 4, 3, 2]) == Some(4);
+    getBy(~f=even, [15, 13, 11]) == None;
+    ```
+  *)
   val getBy : f:('a -> bool) -> 'a list -> 'a option
 
+  (**
+    Same as getBy
```

Comment:
It seems that sometimes you put the doc on the underscore function, and sometimes on the camelCase version. Is there a heuristic to choose them?

Reply:
I went by whatever came first in the file. As to separating the ReasonML and OCaml comments, what do I do for functions like `map` or `any`, which are the same for both?

I have no idea how to rebase out the deleted files. My git workflow usually involves deleting everything, re-cloning, and starting all over.

Reply:
I guess for `map` and `any`, keep them combined? I think the main thing I want to avoid is the separate reason and ocaml tags which makes it very hard to read in the mli. Perhaps "`square n` (`square(n)` in reason)` returns the square of `n`" might be more readable for now?

Reply:
I can do the rebase before merging.

Reply:
For functions like `map` and `any`, I'll use your wording in the description and keep the Ocaml and Reason examples together as they currently are. For something like `get_by`, the description and example will be OCaml only; for `getBy`, the description and example will be ReasonML only. Both of them will have a “Same as <other function>“ note in the description. Does that work for you?

Reply:
Yes, wonderful! And apologies for the back and forth on this - excited to get this merged :)

Reply:
Not a problem, I will fix what I have so far and then commit it so you can see if we are on track before going through all 200+ functions :)

Reply:
Fantastic, thanks again!